hordes-ui-mod.js

```diff
@@ -382,941 +391,1065 @@
+                    $customSettingsWindow.parentNode.removeChild($customSettingsWindow);
+                });
+            });
+        },
+
+        // This creates the initial chat context menu DOM (which starts as hidden)
+        function createChatContextMenu() {
+            if (document.querySelector('.js-chat-context-menu')) {
+                return;
+            }
+
+            let contextMenuHTML = `
     			<div class="js-name">...</div>
 				<div class="choice" name="party">Party invite</div>
 				<div class="choice" name="whisper">Whisper</div>
 				<div class="choice" name="friend">Friend</div>
 				<div class="choice" name="unfriend">Unfriend</div>
 				<div class="choice" name="block">Block</div>
 			`
-    		document.body.appendChild(makeElement({
-    			element: 'div',
-    			class: 'panel context border grey js-chat-context-menu',
-    			content: contextMenuHTML,
-    		}));
-
-    		const $chatContextMenu = document.querySelector('.js-chat-context-menu');
-    		$chatContextMenu.querySelector('[name="party"]').addEventListener('click', () => {
-				modHelpers.partyPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="whisper"]').addEventListener('click', () => {
-    			modHelpers.whisperPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="friend"]').addEventListener('click', () => {
-    			modHelpers.friendPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="unfriend"]').addEventListener('click', () => {
-    			modHelpers.unfriendPlayer(tempState.chatName);
-    		});
-    		$chatContextMenu.querySelector('[name="block"]').addEventListener('click', () => {
-    			modHelpers.blockPlayer(tempState.chatName);
-    		});
-    	},
-
-    	// This opens a context menu when you click a user's name in chat
-    	function chatContextMenu() {
-    		const addContextMenu = ($name, name) => {
-    			$name.classList.add('js-is-context-menu-initd');
-    			// Add name to element so we can target it in CSS when filtering chat for block list
-    			$name.setAttribute('data-chat-name', name);
-
-    			const showContextMenu = clickEvent => {
-    				// TODO: Is there a way to pass the name to showChatContextMenumethod, instead of storing in tempState?
-    				tempState.chatName = name;
-    				modHelpers.showChatContextMenu(name, {x: clickEvent.pageX, y: clickEvent.pageY});
-    			};
-    			$name.addEventListener('click', showContextMenu); // Left click
-    			$name.addEventListener('contextmenu', showContextMenu); // Right click works too
-    		};
-    		Array.from(document.querySelectorAll('.name:not(.js-is-context-menu-initd)')).forEach(($name) => {
-    			addContextMenu($name, $name.textContent);
-    		});
-    		Array.from(document.querySelectorAll('.textwhisper .textf1:not(.js-is-context-menu-initd)')).forEach($whisperName => {
-    			// $w
```

Comment:
Can you add a custom class, e.g. `js-xpmeter-icon`, to this `btn.black.svgicon` we define in the HTML above? Then we can just target that class here instead of having to dig into our `#xpmeter` DOM.

Reply:
I think you marked the wrong lines here.